src/main/java/io/reactivex/internal/operators/flowable/FlowableDistinct.java

```diff
@@ -44,7 +44,7 @@ protected void subscribeActual(Subscriber<? super T> observer) {
         Collection<? super K> collection;
 
         try {
-            collection = collectionSupplier.call();
+            collection = ObjectHelper.requireNonNull(collectionSupplier.call(), "The returned Collection is null. Null values are generally not allowed in 2.x operators and sources.");
```

Comment:
I think it should even be 'The collectionSupplier returned a null Collection. ..'